morphir/src/org/finos/morphir/util/PrintMDMJson.scala

```diff
@@ -0,0 +1,146 @@
+package org.finos.morphir.util
+
+import _root_.org.finos.morphir.naming.*
+import _root_.org.finos.morphir.datamodel.Label
+import _root_.org.finos.morphir.datamodel.EnumLabel
+import _root_.org.finos.morphir.datamodel.Data
+import _root_.org.finos.morphir.datamodel.Concept
+import org.finos.morphir.datamodel.Data.{Optional, Result}
+import zio.json.ast._
+import zio.json.EncoderOps
+
+object PrintMDMJson {
+
+  def prettyPrint(
+      any: Any,
+      detailLevel: DetailLevel = DetailLevel.BirdsEye,
+      fieldNames: FieldNames = FieldNames.Hide
+  ): String =
+    toJson(any, detailLevel, fieldNames).toJsonPretty
+
+  def compactPrint(
+      any: Any,
+      detailLevel: DetailLevel = DetailLevel.BirdsEye,
+      fieldNames: FieldNames = FieldNames.Hide
+  ): String =
+    toJson(any, detailLevel, fieldNames).toJson
+
+  private def toJson(any: Any, detailLevel: DetailLevel, fieldNames: FieldNames): Json =
+    any match {
+      case name: Name   => Json.Str(name.toTitleCase)
+      case qn: FQName   => Json.Str(if (detailLevel.hideFQNames) qn.localName.toTitleCase else qn.toStringTitleCase)
+      case label: Label => Json.Str(label.value)
+      case v: Data      => dataToJson(v)
+      case v: Concept   => conceptToJson(v, detailLevel)
+      case other        => Json.Str(other.toString)
+    }
+
+  private def dataToJson(data: Data): Json =
```

Comment:
I'm okay moving forward with this as is, but we should revisit making sure this is properly tail recursive by applying the `@tailrec` annotation to check for tail recursion (looking at this by eye I can tell it is almost there but the `Data.Case` is not tail recursive.
see: https://www.baeldung.com/scala/tail-recursion 

That being said we can accept this, but there is a chance with a deep enough structure we could blow the stack.